pilot/pkg/model/push_context.go

```diff
@@ -230,7 +232,7 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 	}
 
 	// Merge the target namespaces
-	if len(first.NamespacesUpdated) > 0 || len(other.NamespacesUpdated) > 0 {
+	if len(first.NamespacesUpdated) > 0 && len(other.NamespacesUpdated) > 0 {
```

Comment:
why did this change?

Reply:
I think it was a bug. When one is empty, we should treat it as all namespaces updated.

Maybe i should do this in a separate pr.